src/client_builder.rs

```diff
@@ -61,41 +142,125 @@ where
         ClientBuilder {
             domain,
             port,
-            starttls: false,
+            mode: ConnectionMode::Auto,
+            #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+            tls_kind: TlsKind::Any,
+            #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+            no_tls_verify: false,
         }
     }
 
+    /// Use no encryption when connecting even if the server supports it.
+    pub fn raw(&mut self) -> &mut Self {
+        self.mode = ConnectionMode::Tcp;
+        self
+    }
+
     /// Use [`STARTTLS`](https://tools.ietf.org/html/rfc2595) for this connection.
     #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
     pub fn starttls(&mut self) -> &mut Self {
-        self.starttls = true;
+        self.mode = ConnectionMode::StartTls;
+        self
+    }
+
+    /// Use TLS for this connection.
+    #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+    pub fn tls(&mut self) -> &mut Self {
+        self.mode = ConnectionMode::Tls;
+        self
+    }
+
+    /// Sets the TLS backend to use for this connection.
+    #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+    pub fn tls_kind(&mut self, kind: TlsKind) -> &mut Self {
+        self.tls_kind = kind;
+        self
+    }
+
+    /// Controls the use of certificate validation.
+    ///
+    /// Defaults to `false`.
+    ///
+    /// # Warning
+    ///
+    /// You should only use this as a last resort
+    ///
+    /// See [`native_tls::TlsConnectorBuilder::danger_accept_invalid_certs`],
+    /// [`native_tls::TlsConnectorBuilder::danger_accept_invalid_hostnames`],
+    /// [`rustls::ClientConfig::dangerous`]
+    #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+    pub fn danger_no_tls_verify(&mut self, no_tls_verify: bool) -> &mut Self {
+        self.no_tls_verify = no_tls_verify;
         self
     }
 
-    /// Return a new [`Client`] using a `native-tls` transport.
+    /// Makes a [`Client`] connection using the native TLS backend.
+    /// Forces TLS unless [`ClientBuilder::starttls()`] was called
```

Comment:
I _think_ rustdoc prefers the first line to have a paragraph break after it. At least iirc, the entire "first paragraph" is considered the "short description", which would include the "Forces TLS" bit as currently written. Same for the method below.